### ISSUE TITLE
Add ability to compress JWT body.

### ISSUE THREAD

**Issue by josebarrueta** [jwtk/jjwt#52]

**Comment by lhazlewood** [comment]
Just some thoughts:

```java
public interface CompressionCodec {
    byte[] compress(String s);
    String uncompress(byte[]);
}
```

```java
CompressionCodec codec = new DeflateCodec();
Jwts.builder().compressWith(codec) ....  .build();
```

```java
public interface CompressionCodecResolver {
    public CompressionCodec resolveCompressionCodec(Header header);
}
```

```java
Jwts.parser().setCompressionCodecResolver(myResolver). .... .parse(compact);
```

Any thoughts on this?

**Comment by josebarrueta** [comment]
I was actually thinking in something simpler an more automated for the user for this first iteration

```
 @Test
 void testCompressStringPayloadWithDeflate() {

        byte[] key = MacProvider.generateKey().getEncoded()

        String payload = "this is my test for a payload"

        String compact = Jwts.builder().setPayload(payload).signWith(SignatureAlgorithm.HS256, key)
                .compressWith(CompressionAlgorithm.DEFLATE).compact()

        def jws = Jwts.parser().setSigningKey(key).parsePlaintextJws(compact)

        String parsed = jws.body

        assertEquals "DEF", jws.header.getCompressionAlgorithm()

        assertEquals "this is my test for a payload", parsed
    }

```

**Comment by josebarrueta** [comment]
after further discussion this is the outcome

single interface (no enums):

```java
public interface CompressionCodec {
   
    public static final CompressionCodec DEFLATE = new DeflateCompressionCodec();

    public static final CompressionCodec GZIP = new GzipCompressionCodec();

    String getAlgorithmName();
    byte[] compress(byte[] byte);
    byte[] decompress(byte[] byte);
}
```
The interface will expose two Default implementations DEFLATE and GZIP.

```java
Jwts.builder().compressWith(CompressionCodec.DEFLATE) ....  .build();
```

And we will provide a default resolver that will know about the default implementation, but developers can provide their own.

```java
public interface CompressionCodecResolver {
    public CompressionCodec resolveCompressionCodec(Header header);
}
```

**Comment by lhazlewood** [comment]
Implemented and in master.  Will be released with 0.6.
